### PR TITLE
fix: --name option for the create-tenant command does not take effect

### LINKED ISSUE #11991: --name option for the create-tenant command does not take effect
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

v0.14.1

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

```sh
cd api
poetry run python -m flask create-tenant --email 'admin@local.com' --name "Custom Tenant" --language 'zh-Hans'
```

### ✔️ Expected Behavior

Will create a new tenant with name `Custom Tenant`.

### ❌ Actual Behavior

Will create a new tenant with name `admin's Workspace`:

image

### PR DESCRIPTION
**Pull request by euxx**
# Summary

Fixes #11991

The issue is:

When run `create_tenant` command, it first registers an account, and then call `create_owner_tenant_if_not_exist` method. （`api/commands.py:565`）

However, a default tenant will be created during the register account process.
```py
    # api/services/account_service.py:815
    def register(
        ...
    ) -> Account:
        create account...

            if FeatureService.get_system_features().is_allow_create_workspace:
                tenant = TenantService.create_tenant(f"{account.name}'s Workspace")
```

Therefore, a tenant will always exist when `create_owner_tenant_if_not_exist` is called,  and won't create a tenant with --name option.

```py
    # api/services/account_service.py:517
    # available_ta will always be true
    def create_owner_tenant_if_not_exist(
        account: Account, name: Optional[str] = None, is_setup: Optional[bool] = False
    ):
        """Check if user have a workspace or not"""
        available_ta = (
            TenantAccountJoin.query.filter_by(account_id=account.id).order_by(TenantAccountJoin.id.asc()).first()
        )

        if available_ta:
            return
```

# Screenshots

Run command:
```sh
cd api
poetry run python -m flask create-tenant --email 'admin@local.com' --name "Custom Tenant" --language 'zh-Hans'
```

| Before | After |
|--------|-------|
|image|image|

# Checklist

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

### CODE REVIEW

File: api/services/account_service.py [hunk 1]
```diff
@@ -793,6 +793,7 @@ def register(
         language: Optional[str] = None,
         status: Optional[AccountStatus] = None,
         is_setup: Optional[bool] = False,
+        create_workspace_required: Optional[bool] = True,
```
Comment (yihong0618) [inline api/services/account_service.py:796]: bool is OK?

Comment (euxx) [reply api/services/account_service.py:796]: Hi, thanks for the review. But sorry, I didn't quite get it what you mean.

Is any issue about bool? Could you please be more specific?

Comment (yihong0618) [reply api/services/account_service.py:796]: create_workspace_required: bool = True,
seems there's no chance  None here?

Comment (euxx) [reply api/services/account_service.py:796]: > create_workspace_required: bool = True, seems there's no chance None here?

Yeah, setting the default value of create_workspace_required to True, which ensures the existing code logic remains unchanged.

Only when the create-tenant command is run, and setting create_workspace_required to False to skip create tenant, which allowing create a tenant with custom name by -- option.

**Review by crazywoola — APPROVED** [review]

**Pull request merged**